mobify/progressive-web-scaffold
# Have a single place in the Redux store to track modal states

This PR removes all modal open flags from individual container states/reducers in favour of a top-level (for now) `modals` branch with its own actions. This enables all modals to be closed on page transitions, and reduces code duplication. 

 **JIRA**: https://mobify.atlassian.net/browse/WEB-1118
 **Linked PRs**: #239 

## Changes
- create modals branch with actions, reducers, and selectors
- port all existing modals to the new store area
- use the new actions and selectors directly from containers.

## How to test-drive this PR
- Run the tests
- Preview Merlin's and see that all modals (including the nav and minicart) open and close correctly.
